### PR TITLE
contrib/labstack/{echo, echo.v4}: add support for noDebugStack

### PR DESCRIPTION
**Pull request by mackjmr**
This PR add's support for noDebugstack for both `contrib/labstack/echo` and `contrib/labstack/echo.v4`

Fixes: #999

### CODE REVIEW

File: contrib/labstack/echo.v4/echotrace.go [hunk 1]
```diff
@@ -38,15 +38,19 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 				tracer.Tag(ext.HTTPURL, request.URL.Path),
 				tracer.Measured(),
 			}
+			finishOptions := []tracer.FinishOption{}
```
Comment (gbbr) [inline contrib/labstack/echo.v4/echotrace.go:41]: Can you define this as below?
```suggestion
			var finishOpts []tracer.FinishOption
```
And please move it above (new) line 49, where it is used, without introducing any new empty lines please.

File: contrib/labstack/echo.v4/echotrace.go [hunk 2]
```diff
@@ -38,15 +38,19 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
...
 
 			if !math.IsNaN(cfg.analyticsRate) {
 				opts = append(opts, tracer.Tag(ext.EventSampleRate, cfg.analyticsRate))
 			}
 			if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(request.Header)); err == nil {
 				opts = append(opts, tracer.ChildOf(spanctx))
 			}
+			if cfg.noDebugStack {
+				finishOptions = []tracer.FinishOption{tracer.NoDebugStack()}
```
Comment (gbbr) [inline contrib/labstack/echo.v4/echotrace.go:50]:
```suggestion
				finishOpts = append(finishOpts, tracer.NoDebugStack())
```

File: contrib/labstack/echo.v4/echotrace.go [hunk 3]
```diff
@@ -61,7 +65,7 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 			// serve the request to the next middleware
 			err := next(c)
 			if err != nil {
-				span.SetTag(ext.Error, err)
+				finishOptions = append(finishOptions, tracer.WithError(err))
```
Comment (gbbr) [inline contrib/labstack/echo.v4/echotrace.go:68]: Why did you change this? Please undo this change. It is not necessary. It should be equal to the previous behaviour.

File: contrib/labstack/echo/echotrace.go [hunk 4]
```diff
@@ -38,23 +38,27 @@ func Middleware(opts ...Option) echo.MiddlewareFunc {
 				tracer.Tag(ext.HTTPURL, request.URL.Path),
 				tracer.Measured(),
 			}
+			finishOptions := []tracer.FinishOption{}
```
Comment (gbbr) [inline contrib/labstack/echo/echotrace.go:41]: Please apply all the same changes here.

File: contrib/labstack/echo.v4/echotrace.go (hunk 3 above)
Comment (mackjmr) [reply contrib/labstack/echo.v4/echotrace.go:68]: The reason I added `finishOptions = append(finishOptions, tracer.WithError(err))` is due to this [line](https://github.com/DataDog/dd-trace-go/blob/v1.34.0/ddtrace/tracer/span.go#L308). If I don't add this, I don't believe that `cfg.NoDebugStack` (from finishOpts) will be taken into account.

This is why I decided to delete `span.SetTag(ext.Error, err)` as it seemed like it was duplicate behaviour to `finishOptions = append(finishOptions, tracer.WithError(err))` ?

**Comment by mackjmr** [comment]
Just made the changes based on the review. Sorry for the delay!

**Review by knusbaum — APPROVED** [review]

**Review by knusbaum — APPROVED** [review]
LGTM.
Thanks @mackjmr
